### PR TITLE
Add support for default environment flag

### PR DESCRIPTION
**Pull request by thisiscab**
Before adding this feature, we would have the metric namespaced with
different values. For instance, we would have `luigi_production` for
production metrics and `luigi_staging` for staging metrics. This caused
a whole lot of problems.

Having this new feature will allow us to set up monitoring on multiple
different environments in parallel thus effectively reducing the number
of metrics, we have to manage.

Relevant CH ticket: https://app.clubhouse.io/glossier-dot-com/story/9091

### CODE REVIEW

**Review by tpalacio — APPROVED** [review]
Adding the environment flag makes a lot of sense. It does introduce the chance of human error where someone can now combine metrics in Datadog for multiple environments if aren't careful. But I definitely feel like this is the right approach. Just talking out loud.

**Comment by thisiscab** [comment]
@hugobast I would still love for you to give me thoughts.

**Comment by thisiscab** [comment]
@hugobast May I receive a review on that, please?

File: luigi/contrib/datadog.py [hunk 1]
```diff
@@ -99,7 +100,13 @@ def _format_task_params_to_tags(self, task):
...
     def default_event_tags(self):
-        if not self._config.default_event_tags:
-            return []
+        default_tags = []
 
-        return str.split(self._config.default_event_tags, ',')
+        if self._config.default_event_tags:
+            default_tags = default_tags + str.split(self._config.default_event_tags, ',')
+
+        if self._config.environment:
```
Comment (hugobast) [inline luigi/contrib/datadog.py]: This one has a default, do you need to `if` check it?

**Review by hugobast — APPROVED** [review]
Looks good, have a small comment

File: luigi/contrib/datadog.py (hunk 1 above)
Comment (thisiscab) [reply luigi/contrib/datadog.py]: You're absolutely right! :)

**Pull request merged**